// provides MessageFile
import "goby/common/protobuf/option_extensions.proto";

package goby.acomms.protobuf;

message DCCLConfig
{
  //  optional int32 modem_id = 1 [(goby.field).description="Unique number 1-31 to identify this node", (goby.field).example="1"];
  optional string crypto_passphrase = 2 [(goby.field).description="If given, encrypt all communications with this passphrase using AES. Omit for unencrypted communications.", (goby.field).example="twinkletoes%24"];

  enum IDCodec
  {
    VARINT = 0;
    LEGACY_CCL = 1;
  }
  optional IDCodec id_codec = 3 [default = VARINT, (goby.field).description="Codec used to represent the unique DCCL identifier (goby.msg).dccl.id, used to match an encoded message to a Protobuf message."];
  
}

